internal/venice-client-common/src/main/java/com/linkedin/venice/compression/ZstdWithDictCompressor.java

```diff
@@ -21,18 +23,18 @@
 
 
 public class ZstdWithDictCompressor extends VeniceCompressor {
-  private ZstdCompressCtx compressor;
-  private byte[] dictionary;
+  private ZstdDictCompress compressorDictionary;
+  private ZstdDictDecompress decompressorDictionary;
 
   public ZstdWithDictCompressor(final byte[] dictionary, int level) {
     super(CompressionStrategy.ZSTD_WITH_DICT);
-    this.dictionary = dictionary;
-    compressor = new ZstdCompressCtx().loadDict(this.dictionary).setLevel(level);
+    this.compressorDictionary = new ZstdDictCompress(dictionary, level);
+    this.decompressorDictionary = new ZstdDictDecompress(dictionary);
   }
 
   @Override
   public byte[] compress(byte[] data) {
-    return compressor.compress(data);
+    return Zstd.compress(data, compressorDictionary);
```

Comment:
```
public static byte[] compress(byte[] src, ZstdDictCompress dict) throws ZstdException {
        ZstdCompressCtx ctx = new ZstdCompressCtx();
        try {
            ctx.loadDict(dict);
            ctx.setLevel(dict.level());
            return ctx.compress(src);
        } finally {
            ctx.close();
        }
    }
```
This method will construct a new instance of `ZstdCompressCtx` and load the dict every time, and I am wondering whether this could be too heavy or not.
Is it possible to run some benchmark before committing this code?
If it is too expensive, can we use thread-local variables to prevent race condition?

Reply:
Here is the javadoc of :`ZstdCompressCtx`
```
Create a context for faster compress operations One such context is required for each thread - put this in a ThreadLocal.
```

It seems using `ThreadLocal` is the right way to achieve best performance without race condition, so can you double check?

Reply:
That's a great idea